pkg/datapath/loader/compile.go

```diff
@@ -379,7 +377,8 @@ func compileDatapath(ctx context.Context, dirs *directoryInfo, isHost, debug boo
 
 // Compile compiles a BPF program generating an object file.
 func Compile(ctx context.Context, src string, out string) error {
-	debug := option.Config.BPFCompilationDebug
+	debug := option.Config.Opts.IsEnabled(option.DebugCompile)
```

Comment:
As we need to keep the deprecated option until 1.11, this needs to be `debug := option.Config.BPFCompilationDebug || option.Config.Opts.IsEnabled(option.DebugCompile)`